src/init/variables/extractors/BaseUrlExtractor.ts

```diff
@@ -22,6 +22,11 @@ export class BaseUrlExtractor extends ShorthandExtractor {
       throw new Error('BaseUrl argument should be provided when using Unix Domain Sockets.');
     }
     const port = args.port ?? this.defaultPort;
+    // Node URL, and potentially other libraries, remove port 80 from the URL when doing URL manipulations,
+    // causing issues when comparing to the stored base URL that does contain port 80.
+    if (port === 80) {
+      return 'http://localhost/';
+    }
     return `http://localhost:${port}/`;
```

Comment:
Let's just not do the formatting ourselves.

```suggestion
    const url = Object.assign(new URL('http://localhost/'), { port });
    return url.href;
```